shared/events.js

```diff
@@ -348,16 +362,50 @@ export class PostMessage extends HashableAction {
   static fields = PostMessage.Fields([
     ['from', 'string'],
     ['headers', 'string', 'repeated'],
-    ['message', 'string']
+    ['message', 'string'],
+    ['sentDate', 'string'],
+    ['read', 'bool']
   ])
+  constructor (data: JSONObject, parentHash?: string) {
+    super({...data, read: false}, parentHash)
+  }
 }
-
 export class PostInvite extends HashableAction {
   static fields = PostInvite.Fields([
-    ['groupId', 'string']
+    ['groupId', 'string'],
+    ['sentDate', 'string'],
+    ['read', 'bool']
+  ])
+  constructor (data: JSONObject, parentHash?: string) {
+    super({...data, read: false}, parentHash)
+  }
+}
+export class RecordInvitation extends HashableAction {
+  static fields = RecordInvitation.Fields([
+    ['username', 'string'],
+    ['inviteHash', 'string'],
+    ['sentDate', 'string']
+  ])
+}
```

Comment:
Record Invitation is a record in the Group Contract that an invite has been sent.